### PR TITLE
readme: explain L4V_ARCH and spec generation

### PR DESCRIPTION
**Pull request by lsf37**
We have so far not been mentioning L4V_ARCH in the instructions and haven't pointed out which sessions need generated input.

Add this information to the instructions.

(supersedes #585)

### CODE REVIEW

**Comment by lsf37** [comment]
@michaelmcinerney are these instructions helpful or still too obscure?

**Comment by michaelmcinerney** [comment]
> @michaelmcinerney are these instructions helpful or still too obscure?

I think that is helpful. I thought that we needed to set the arch to be `RISCV64` to generate the `Kernel_Config` file (for some reason I didn't understand) but if that's not the case, then great!

File: README.md [hunk 1]
```diff
@@ -165,44 +165,55 @@ Running the Proofs
...
 from the directory `l4v/`.
 
-Not all of the proof sessions can be built directly with the `isabelle build` command.
-The seL4 verification proofs depend on Isabelle specifications that are
-generated from the C source code and Haskell model.
-Therefore, it's recommended to always build using the supplied makefiles,
-which will ensure that these generated specs are up to date.
+Set the environment variable `L4V_ARCH` to one of `ARM`, `ARM_HYP`, `X64`,
+`RISCV64`, or `AARCH64` to get the proofs for the respective architecture. `ARM`
+has the most complete set of proofs, the other architectures tend support only a
```
Comment (michaelmcinerney) [inline README.md:176]: "tend support" --> "tend to support"

**Comment by lsf37** [comment]
> > @michaelmcinerney are these instructions helpful or still too obscure?
> 
> I think that is helpful. I thought that we needed to set the arch to be `RISCV64` to generate the `Kernel_Config` file (for some reason I didn't understand) but if that's not the case, then great!

Cool. No, it just needs to be set to the architecture you want to run the proofs for. In your case that was RISCV64.

File: README.md [hunk 2]
```diff
@@ -165,44 +165,55 @@ Running the Proofs
...
+For example, to build the abstract specification, do
 
+    export L4V_ARCH=ARM
     cd l4v/spec
-    make ExecSpec
+    make ASpec
 
 See the `HEAPS` variable in the corresponding `Makefile` for available targets.
+The sessions that directly depend on generated sources are `ASpec`, `ExecSpec`,
+and `CKernel`. These, and all sessions that depend on them need to be run using
```
Comment (michaelmcinerney) [inline README.md:194]: Add a comma after "them"?

File: README.md [hunk 3]
```diff
@@ -165,44 +165,55 @@ Running the Proofs
...
-`ROOT` files in this repository. There is roughly one session corresponding to
-each major directory in the repository.
+from the directory `l4v/`. For available sessions and their dependencies, see
+the corresponding `ROOT` files in this repository. There is roughly one session
+corresponding to each major directory in the repository.
 
 For interactively exploring, say the invariant proof of the abstract
-specification with a pre-built logic image for the abstract specification and
-all of the invariant proof's dependencies, run
+specification on `ARM`, note that in `proofs/ROOT` the parent session for
```
Comment (michaelmcinerney) [inline README.md:206]: `proof` instead of `proofs`

**Review by michaelmcinerney — APPROVED** [review]
👍

**Pull request merged**